server/core_benchmarks_test.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2023 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !skip_js_tests && !skip_js_cluster_tests && !skip_js_cluster_tests_2
+// +build !skip_js_tests,!skip_js_cluster_tests,!skip_js_cluster_tests_2
+
+package server
+
+import (
+	"crypto/rand"
+	"crypto/tls"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/nats-io/nats.go"
+)
+
+func BenchmarkTLSReqReply(b *testing.B) {
+	const (
+		subject    = "test-subject"
+		configPath = "./configs/tls"
+	)
+
+	// default TLS client connection options
+	defaultOpts := []nats.Option{
+		nats.Secure(
+			&tls.Config{
+				InsecureSkipVerify: true,
+			},
+		),
+	}
+
+	schemes := []string{
+		"none",
+		"ed25519",
+		"rsa-1024",
+		"rsa-2048",
+		"rsa-4096",
+	}
+	payloadSzs := []int64{
+		1024,   // 1kb
+		4096,   // 4kb
+		40960,  // 40kb
+		409600, // 400kb
+	}
+
+	for _, scheme := range schemes {
+		schemeConfig := fmt.Sprintf("%s/tls-%s.conf", configPath, scheme)
+		b.Run(fmt.Sprintf("tlsScheme=%s", scheme), func(b *testing.B) {
+			for _, payloadSz := range payloadSzs {
+				b.Run(fmt.Sprintf("payloadSz=%db", payloadSz), func(b *testing.B) {
+					var (
+						server *Server
+						opts   []nats.Option
+					)
+					// no TLS
+					if scheme == "none" {
+						server = RunServer(&DefaultTestOptions)
+						opts = []nats.Option{}
+					} else {
+						// run server with tls scheme
+						server, _ = RunServerWithConfig(schemeConfig)
```

Comment:
Any chance you can start the `none` server using the same method?

Reply:
I couldn't find a config that just had a `listen: x.x.x.x:XXXX` field. I was debating making a `certs/tls/none.conf` file but I wanted to save space by not including it. What do you think?

Reply:
I think it's fair to add as `tls/none.conf` for symmetry in tests like this one